src/common/components/map/d3Components/map.ts

```diff
@@ -0,0 +1,148 @@
+import { select, BaseType } from 'd3';
+import { geoPath, GeoProjection, GeoPath, GeoPermissibleObjects } from 'd3-geo';
+import { GeometryObject, FeatureCollection, MultiLineString } from 'geojson';
+import { D3Selection, Extension } from '../../../types';
+import { GeoArea } from '../viewModel';
+import { SVG_SHADOW_ID } from './constants';
+import { shadowDefinitions } from './shadowDefinitions';
+import { tooltipComponent } from '../../../d3Components/tooltip';
+import { zoomComponent } from '../../../d3Components/zoom';
+const styles = require('./map.scss');
+
+interface Props {
+  root: D3Selection;
+  svg: D3Selection;
+  geoAreas: GeoArea[];
+  geoEntities: FeatureCollection<GeometryObject, any>;
+  mesh: MultiLineString;
+  projection: GeoProjection;
+  width: number;
+  height: number;
+  padding: number;
+  defaultfillColor: string;
+  maxZoomScale: number;
+  clickZoomFitScale: number;
+  highlightColor?: string
+}
+
+interface State {
+  map: D3Selection;
+  mapExtension: Extension;
+  geoPathGenerator: GeoPath<any, GeoPermissibleObjects>;
+  onShowTooltip: (message: string) => void;
+  onHideTooltip: () => void;
+  onUpdateTooltipPosition: () => void;
+}
+
+export const mapComponent = (props: Props) => {
+  const state = createEmptyState();
+  shadowDefinitions({
+    svg: props.svg,
+  });
+
+  state.map = renderMap(props);
+  state.mapExtension = calculateMapExtension(props);
+  state.geoPathGenerator = getGeoPathGenerator(props, state);
+
+  enter(props, state);
+  renderMesh(props, state);
+
+  zoomComponent({
+    svg: props.svg,
+    node: state.map,
+    zoomAreas: props.geoAreas,
+    nodeExtension: state.mapExtension,
+    nodeSelectionElement: 'path',
+    getZoomAreaExension: (geoArea: GeoArea) => (
+      state.geoPathGenerator.bounds(geoArea.geoEntity)
+    ),
+    maxZoomScale: props.maxZoomScale,
+    clickZoomFitScale: props.clickZoomFitScale,
+    width: props.width,
+    height: props.height,
+  });
+
+  const { onShowTooltip, onHideTooltip, onUpdateTooltipPosition } = tooltipComponent({
+    node: props.root,
+  });
+  state.onShowTooltip = onShowTooltip;
+  state.onHideTooltip = onHideTooltip;
+  state.onUpdateTooltipPosition = onUpdateTooltipPosition;
+};
+
+const createEmptyState = (): State => ({
+  map: null,
+  mapExtension: null,
+  geoPathGenerator: null,
+  onShowTooltip: null,
+  onHideTooltip: null,
+  onUpdateTooltipPosition: null,
+});
```

Comment:
Shoudl we create a map.business.ts and extract props and state and functions like this? _createEmptyState_ not sure if it would be good or not, depends if more isolated functions can be extracted enhancing readiblity.